mesh_generation/generate_mesh.py

```diff
@@ -29,35 +31,46 @@
 import pandas as pd
 
 
-def is_git_repo():
+def get_git_url(file):
     """
-    Return True/False depending on whether or not the current directory is a git repo.
+    If the provided file is in a git repo, return the url to its most recent commit remote.origin.
     """
+    dirname = os.path.dirname(file)
 
-    return subprocess.call(
-        ['git', '-C', '.', 'status'],
-        stderr=subprocess.STDOUT,
-        stdout = open(os.devnull, 'w')
-    ) == 0
+    try:
+        url = subprocess.check_output(
+            ["git", "-C", dirname, "config", "--get", "remote.origin.url"]
+        ).decode("ascii").strip()
+        url = url.removesuffix('.git')
+    except subprocess.CalledProcessError:
+        return None
 
-def git_info():
-    """
-    Return the git repo origin url, relative path to this file, and latest commit hash.
-    """
+    if url.startswith("git@github.com:"):
+        url = f"https://github.com/{url.removeprefix('git@github.com:')}"
 
-    url = subprocess.check_output(
-        ["git", "remote", "get-url", "origin"]
-    ).decode('ascii').strip()
     top_level_dir = subprocess.check_output(
-        ['git', 'rev-parse', '--show-toplevel']
-    ).decode('ascii').strip()
-    rel_path = os.path.relpath(__file__, top_level_dir)
+        ["git", "-C", dirname, "rev-parse", "--show-toplevel"]
+    ).decode("ascii").strip()
+    rel_path = file.removeprefix(top_level_dir)
+
     hash = subprocess.check_output(
-        ['git', 'rev-parse', 'HEAD']
-    ).decode('ascii').strip()
+        ["git", "-C", dirname, "rev-parse", "HEAD"]
+    ).decode("ascii").strip()
 
-    return url, rel_path, hash
+    return f"{url}/blob/{hash}{rel_path}"
```

Comment:
or maybe
```suggestion
    return f"{url}/tree/{hash}{rel_path}"
```
? I don't really know the difference, so feel free to ignore...

Reply:
> 

Neither did I. Apparently [`blob` is for files, `tree` is for directories](https://stackoverflow.com/questions/39400848/in-github-urls-what-is-the-difference-between-a-tree-and-a-blob). But the browser redirects to whichever is correct, so it doesn't really matter. I'll leave as `blob` since this is a file.